cpp/arcticdb/storage/mongo/mongo_client.cpp

```diff
@@ -310,31 +360,28 @@ storage::KeySegmentPair MongoClientImpl::read_segment(const std::string &databas
     auto database = client->database(database_name); //TODO maybe cache
     auto collection = database[collection_name];
 
-    try {
-        ARCTICDB_SUBSAMPLE(MongoStorageReadFindOne, 0)
-        auto stream_id = variant_key_id(key);
-        if(StorageFailureSimulator::instance()->configured())
-            StorageFailureSimulator::instance()->go(FailureType::READ);
+    ARCTICDB_SUBSAMPLE(MongoStorageReadFindOne, 0)
+    auto stream_id = variant_key_id(key);
+    if(StorageFailureSimulator::instance()->configured())
+        StorageFailureSimulator::instance()->go(FailureType::READ);
 
+    try {
         auto result = collection.find_one(document{} << "key" << fmt::format("{}", key) << "stream_id" <<
-                                                                                                       fmt::format("{}", stream_id) << finalize);
+                                                     fmt::format("{}", stream_id) << finalize);
         if (result) {
-            const auto &doc = result->view();
+            const auto& doc = result->view();
             auto size = doc["total_size"].get_int64().value;
-            entity::VariantKey stored_key{ detail::variant_key_from_document(doc, key) };
+            entity::VariantKey stored_key{detail::variant_key_from_document(doc, key)};
             util::check(stored_key == key, "Key mismatch: {} != {}");
```

Comment:
This is a general question not related to this PR in particular. Do we want to leak `E_ASSERTION_FAILURE` exceptions? AFAIK we use this in places where people would use cassert's function `assert` so do we want people to be able to see this in "raw" form and why don't we use casserts in general?